lib/directory-view.coffee

```diff
@@ -108,10 +108,12 @@ class DirectoryView
       view = new FileView(entry)
 
     subscription = @directory.onDidRemoveEntries (removedEntries) ->
-      for removedName, removedEntry of removedEntries when entry is removedEntry
+      removedEntries.forEach (removedEntry, removedName) ->
+        return unless entry is removedEntry
         view.element.remove()
         subscription.dispose()
-        break
+        # TODO: When this file is converted to JS, convert this forEach loop
+        # to a for...of loop and add a break here for performance
```

Comment:
Yep, it's annoying because Coffeescript's `for...of` is JS's `for...in`, and of course its `for...in` is just a fancy for loop and not `for...of`.  So there's no way to loop Map entries.  Oh well.